server/src/main/java/com/continuuity/loom/store/cluster/BaseSQLClusterStoreView.java

```
@@ -0,0 +1,222 @@
+      Connection conn = dbConnectionPool.getConnection();
+      try {
+        PreparedStatement statement = getClusterExistsStatement(conn, clusterNum);
+        try {
+          ResultSet rs = statement.executeQuery();
+          try {
+            return rs.next();
+          } finally {
+            rs.close();
+          }
+        } finally {
+          statement.close();
+        }
+      } finally {
+        conn.close();
+      }
+    } catch (SQLException e) {
+      throw new IOException("Exception checking existence of cluster " + clusterId, e);
+    }
+  }
+
+  @Override
+  public void writeCluster(Cluster cluster) throws IllegalAccessException, IOException {
+    if (!allowedToWrite(cluster)) {
+      throw new IllegalAccessException("Not allowed to write cluster " + cluster.getId());
+    }
+    long clusterNum = Long.parseLong(cluster.getId());
+    // sticking with standard sql... this could be done in one step with replace, or with
+    // insert ... on duplicate key update with mysql.
+    try {
+      Connection conn = dbConnectionPool.getConnection();
+      try {
+        PreparedStatement writeStatement;
+        ByteArrayInputStream clusterBytes = new ByteArrayInputStream(codec.serialize(cluster, Cluster.class));
+        if (clusterExists(cluster.getId())) {
+          writeStatement = getSetClusterStatement(conn, clusterNum, cluster, clusterBytes);
+        } else {
+          writeStatement = getInsertClusterStatement(conn, clusterNum, cluster, clusterBytes);
+        }
+
+        // perform the update or insert
+        try {
+          writeStatement.executeUpdate();
+        } finally {
+          writeStatement.close();
+        }
+      } finally {
+        conn.close();
+      }
+    } catch (SQLException e) {
+      throw new IOException("Exception writing cluster " + cluster.getId(), e);
+    }
+  }
+
+  @Override
+  public void deleteCluster(String clusterId) throws IOException {
+    long clusterNum = Long.parseLong(clusterId);
```

writeCluster throws an exception when a user is not allowed to write to a cluster, but deleteCluster silently ignores it. We need to make the behavior uniform.